experimental/golden_response_checker.go

```diff
@@ -0,0 +1,122 @@
+package experimental
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// CheckGoldenDataResponse will verify that the stored file matches the given data.DataResponse
+// when the updateFile flag is set, this will both add errors to the response and update the saved file
+func CheckGoldenDataResponse(path string, dr *backend.DataResponse, updateFile bool) error {
+	saved, err := readGoldenFile(path)
+	if err != nil {
+		err = fmt.Errorf("error reading golden file:  %s\n%s", path, err.Error())
```

Comment:
return fmt.Errorf() and then everything else can be outdented.

Reply:
interesting -- I added a utility function that will check the updateFlag and then return the error.  This makes the indenting much nicer.